ui/js/compareperf.js

```diff
@@ -5,13 +5,46 @@
 "use strict";
 
 perf.controller('CompareChooserCtrl', [
-  '$state', '$stateParams', '$scope', 'ThRepositoryModel', 'ThResultSetModel',
-  function CompareChooserCtrl($state, $stateParams, $scope,
+  '$state', '$stateParams', '$scope', '$http', 'ThRepositoryModel', 'ThResultSetModel',
+  function CompareChooserCtrl($state, $stateParams, $scope, $http,
                               ThRepositoryModel, ThResultSetModel) {
     ThRepositoryModel.get_list().success(function(projects) {
       $scope.projects = projects;
       $scope.originalProject = $scope.newProject = projects[0];
+      $scope.oriTipList = [];
+      $scope.newTipList = [];
+      var getRevisionTips = function (projectName, list) {
+        // due to we push the revision data into list,
+        // so we need clear the data before we push new data into it.
+        list.splice(0, list.length);
+        return $http.get (thServiceDomain + '/api/project/' +
```

Comment:
Also, there should be no space between a function call (or declaration) and the parentheses:

i.e. always do:

    function foo(bar) { ...
    }
    foo(123);

and not:

    function foo (bar) { ... // WRONG
    }
    foo (123); // WRONG